.github/workflows/lint.yaml

```diff
@@ -13,39 +13,84 @@ jobs:
         with:
           fetch-depth: 1
 
-      - name: Cache node
-        uses: actions/cache@v2
-        with:
-          path: ~/.npm
-          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
-          restore-keys: |
-            ${{ runner.os }}-node-
-
-      - name: Setup Node.js
+      - name: Setup node.js
         uses: actions/setup-node@v1
         with:
           node-version: '12.x'
 
-      - name: Install dependencies
-        run: yarn
+      - name: Set yarn cache directory path
+        id: yarn-cache-dir-path
+        run: echo "::set-output name=dir::$(yarn cache dir)"
+
+      - name: Restore yarn cache
+        uses: actions/cache@v2
+        id: yarn-cache
+        with:
+          # mind node_modules backup
+          path: |
+            **/node_modules
+            ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+            ${{ runner.os }}-yarn-
+
+      - name: Install node.js dependencies
+        if: steps.yarn-cache.outputs.cache-hit != 'true'
+        run: yarn --frozen-lockfile
 
       - name: Run linter on *.sol and *.json
-        run: yarn run lint:check
+        run: yarn lint:check
+
+    commits:
+      runs-on: ubuntu-latest
+
+      steps:
+      - name: Check out github repository
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Run commitlint
+        uses: wagoid/commitlint-github-action@v2
 
     brownie:
       runs-on: ubuntu-latest
+
       steps:
-      - uses: actions/checkout@v1
+      - name: Check out github repository
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 1
 
-      - name: Set up Python 3.8
-        uses: actions/setup-python@v1
+      - name: Set up python 3.8
+        uses: actions/setup-python@v2
         with:
           python-version: 3.8
 
-      - name: Install black
+      - name: Set pip cache directory path
+        id: pip-cache-dir-path
+        run: |
+            echo "::set-output name=dir::$(pip cache dir)"
+
+      - name: Restore pip cache
+        uses: actions/cache@v2
+        id: pip-cache
+        with:
+          # mind Linux runner
+          path: |
+            /opt/hostedtoolcache/Python/3.8.*
+            ${{ steps.pip-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-pip-${{ hashFiles('**/requirements-dev.txt') }}
+          restore-keys: |
+            ${{ runner.os }}-pip-${{ hashFiles('**/requirements-dev.txt') }}
+            ${{ runner.os }}-pip-
+
+      - name: Install python dependencies
+        if: steps.pip-cache.outputs.cache-hit != 'true'
         run: pip install -r requirements-dev.txt
 
-      - name: Run Black
-        run: black .
+      - name: Run black
+        run: black tests --check --config black-config.toml
```

Comment:
when running `black .` locally with your branch I get:
```
reformatted /Users/csessa/dev/yearn-protocol/tests/functional/controllers/test_config.py
reformatted /Users/csessa/dev/yearn-protocol/tests/functional/utils/test_oracles.py
All done! ✨ 🍰 ✨
2 files reformatted, 6 files left unchanged.
```

Shouldn't this be `black --check --config black-config.toml`?

I also couldn't find any documentation on `tests` 🤷

Reply:
Created a separate issue to add README. To check formatting with black locally you can run `black tests --check --config black-config.toml`.

Reply:
@milkyklim oh, plz dismiss my comment, I confused `tests` as if it was a command inside black but it's actually the folder holding the python code 🤦

Reply:
I actually think @Macarse's suggestion is good, when we add stuff to `scripts/` it would be good to have it cover that as well